pysetup/spec_builders/deneb.py

```diff
@@ -63,9 +69,22 @@ def verify_and_notify_new_payload(self: ExecutionEngine,
 
 
     @classmethod
-    def hardcoded_custom_type_dep_constants(cls, spec_object) -> str:
+    def hardcoded_custom_type_dep_constants(cls, spec_object) -> Dict[str, str]:
         return {
             'BYTES_PER_FIELD_ELEMENT': spec_object.constant_vars['BYTES_PER_FIELD_ELEMENT'].value,
             'FIELD_ELEMENTS_PER_BLOB': spec_object.preset_vars['FIELD_ELEMENTS_PER_BLOB'].value,
             'MAX_BLOBS_PER_BLOCK': spec_object.preset_vars['MAX_BLOBS_PER_BLOCK'].value,
+            'MAX_BLOB_COMMITMENTS_PER_BLOCK': spec_object.preset_vars['MAX_BLOB_COMMITMENTS_PER_BLOCK'].value,
+        }
+
+    @classmethod
+    def hardcoded_ssz_dep_constants(cls) -> Dict[str, str]:
+        return {
+            'BLOB_KZG_COMMITMENTS_GINDEX': 'GeneralizedIndex(27)',
+        }
+
+    @classmethod
+    def hardcoded_func_dep_presets(cls, spec_object) -> Dict[str, str]:
+        return {
+            'KZG_COMMITMENT_INCLUSION_PROOF_DEPTH': spec_object.preset_vars['KZG_COMMITMENT_INCLUSION_PROOF_DEPTH'].value,
```

Comment:
Note to other reviewers: `KZG_COMMITMENT_INCLUSION_PROOF_DEPTH` is tricky because (1) it depends on another preset value `MAX_BLOB_COMMITMENTS_PER_BLOCK` and (2) it needs helper `floorlog2` and `ceillog2`.

To ensure that the preset setting is equal to the markdown definition result, I did some dirty trick around "`func_dep_presets`" to add the assertion `assert KZG_COMMITMENT_INCLUSION_PROOF_DEPTH == uint64(floorlog2(BLOB_KZG_COMMITMENTS_GINDEX) + 1 + ceillog2(MAX_BLOB_COMMITMENTS_PER_BLOCK))` at the end of pyspec (`minimal.py` and `mainnet.py`).

It doesn't look pretty. 😅 Open to other simpler solutions.

Reply:
A unit test would be good enough?

Reply:
Yeah, I only added `test_config_invariants` unit test when checking `MAX_REQUEST_BLOB_SIDECARS`. `func_dep_presets` is a less-duplicate code solution though.

Reply:
Alternatively, `MAX_KZG_COMMITMENT_INCLUSION_PROOF_LENGTH` can be defined and set to a bigger value e.g. `20` to accommodate larger proofs for the case if BeaconBlockBody is extended in the future.

The depth for validation purposes can be derived from the gindex as it is already done in the code.